// @twurple/eventsub-base/lib/events/moderation/EventSubChannelUnvipModerationEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventSubChannelUnvipModerationEvent = void 0;
const tslib_1 = require("tslib");
const common_1 = require("@twurple/common");
const EventSubChannelBaseModerationEvent_1 = require("./EventSubChannelBaseModerationEvent");
/**
 * An EventSub event representing a user having lost VIP status on a channel.
 */
let EventSubChannelUnvipModerationEvent = class EventSubChannelUnvipModerationEvent extends EventSubChannelBaseModerationEvent_1.EventSubChannelBaseModerationEvent {
    constructor() {
        super(...arguments);
        this.moderationAction = 'unvip';
    }
    /**
     * The ID of the user losing VIP status.
     */
    get userId() {
        return this[common_1.rawDataSymbol].unvip.user_id;
    }
    /**
     * The name of the user losing VIP status.
     */
    get userName() {
        return this[common_1.rawDataSymbol].unvip.user_login;
    }
    /**
     * The display name of the user losing VIP status.
     */
    get userDisplayName() {
        return this[common_1.rawDataSymbol].unvip.user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return await this._client.users.getUserById(this[common_1.rawDataSymbol].unvip.user_id);
    }
};
exports.EventSubChannelUnvipModerationEvent = EventSubChannelUnvipModerationEvent;
exports.EventSubChannelUnvipModerationEvent = EventSubChannelUnvipModerationEvent = tslib_1.__decorate([
    (0, common_1.rtfm)('eventsub-base', 'EventSubChannelUnvipModerationEvent', 'broadcasterId')
], EventSubChannelUnvipModerationEvent);
