// @twurple/eventsub-base/lib/events/moderation/EventSubChannelBaseModerationEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventSubChannelBaseModerationEvent = void 0;
const tslib_1 = require("tslib");
const common_1 = require("@twurple/common");
const shared_utils_1 = require("@d-fischer/shared-utils");
let EventSubChannelBaseModerationEvent = class EventSubChannelBaseModerationEvent extends common_1.DataObject {
    /** @internal */
    constructor(data, client) {
        super(data);
        this._client = client;
    }
    /**
     * The ID of the broadcaster.
     */
    get broadcasterId() {
        return this[common_1.rawDataSymbol].broadcaster_user_id;
    }
    /**
     * The name of the broadcaster.
     */
    get broadcasterName() {
        return this[common_1.rawDataSymbol].broadcaster_user_login;
    }
    /**
     * The display name of the broadcaster.
     */
    get broadcasterDisplayName() {
        return this[common_1.rawDataSymbol].broadcaster_user_name;
    }
    /**
     * Gets more information about the broadcaster.
     */
    async getBroadcaster() {
        return await this._client.users.getUserById(this[common_1.rawDataSymbol].broadcaster_user_id);
    }
    /**
     * The ID of the moderator who performed the action.
     */
    get moderatorId() {
        return this[common_1.rawDataSymbol].moderator_user_id;
    }
    /**
     * The name of the moderator who performed the action.
     */
    get moderatorName() {
        return this[common_1.rawDataSymbol].moderator_user_login;
    }
    /**
     * The display name of the moderator who performed the action.
     */
    get moderatorDisplayName() {
        return this[common_1.rawDataSymbol].moderator_user_name;
    }
    /**
     * Gets more information about the broadcaster.
     */
    async getModerator() {
        return await this._client.users.getUserById(this[common_1.rawDataSymbol].moderator_user_id);
    }
};
exports.EventSubChannelBaseModerationEvent = EventSubChannelBaseModerationEvent;
tslib_1.__decorate([
    (0, shared_utils_1.Enumerable)(false)
], EventSubChannelBaseModerationEvent.prototype, "_client", void 0);
exports.EventSubChannelBaseModerationEvent = EventSubChannelBaseModerationEvent = tslib_1.__decorate([
    (0, common_1.rtfm)('eventsub-base', 'EventSubChannelBaseModerationEvent', 'broadcasterId')
], EventSubChannelBaseModerationEvent);
