// @twurple/eventsub-base/lib/events/moderation/EventSubChannelSharedChatDeleteModerationEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventSubChannelSharedChatDeleteModerationEvent = void 0;
const tslib_1 = require("tslib");
const common_1 = require("@twurple/common");
const EventSubChannelBaseModerationEvent_1 = require("./EventSubChannelBaseModerationEvent");
/**
 * An EventSub event representing a moderator deleting a message in another channel during a shared chat session.
 */
let EventSubChannelSharedChatDeleteModerationEvent = class EventSubChannelSharedChatDeleteModerationEvent extends EventSubChannelBaseModerationEvent_1.EventSubChannelBaseModerationEvent {
    constructor() {
        super(...arguments);
        this.moderationAction = 'shared_chat_delete';
    }
    /**
     * The ID of the user whose message is being deleted.
     */
    get userId() {
        return this[common_1.rawDataSymbol].shared_chat_delete.user_id;
    }
    /**
     * The name of the user whose message is being deleted.
     */
    get userName() {
        return this[common_1.rawDataSymbol].shared_chat_delete.user_login;
    }
    /**
     * The display name of the user whose message is being deleted.
     */
    get userDisplayName() {
        return this[common_1.rawDataSymbol].shared_chat_delete.user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return await this._client.users.getUserById(this[common_1.rawDataSymbol].shared_chat_delete.user_id);
    }
    /**
     * The ID of the message being deleted.
     */
    get messageId() {
        return this[common_1.rawDataSymbol].shared_chat_delete.message_id;
    }
    /**
     * The message text of the message being deleted.
     */
    get messageText() {
        return this[common_1.rawDataSymbol].shared_chat_delete.message_body;
    }
};
exports.EventSubChannelSharedChatDeleteModerationEvent = EventSubChannelSharedChatDeleteModerationEvent;
exports.EventSubChannelSharedChatDeleteModerationEvent = EventSubChannelSharedChatDeleteModerationEvent = tslib_1.__decorate([
    (0, common_1.rtfm)('eventsub-base', 'EventSubChannelSharedChatDeleteModerationEvent', 'broadcasterId')
], EventSubChannelSharedChatDeleteModerationEvent);
