// @twurple/eventsub-base/lib/events/moderation/EventSubChannelUnmodModerationEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventSubChannelUnmodModerationEvent = void 0;
const tslib_1 = require("tslib");
const common_1 = require("@twurple/common");
const EventSubChannelBaseModerationEvent_1 = require("./EventSubChannelBaseModerationEvent");
/**
 * An EventSub event representing a user having lost moderator status on a channel.
 */
let EventSubChannelUnmodModerationEvent = class EventSubChannelUnmodModerationEvent extends EventSubChannelBaseModerationEvent_1.EventSubChannelBaseModerationEvent {
    constructor() {
        super(...arguments);
        this.moderationAction = 'unmod';
    }
    /**
     * The ID of the user losing mod status.
     */
    get userId() {
        return this[common_1.rawDataSymbol].unmod.user_id;
    }
    /**
     * The name of the user losing mod status.
     */
    get userName() {
        return this[common_1.rawDataSymbol].unmod.user_login;
    }
    /**
     * The display name of the user losing mod status.
     */
    get userDisplayName() {
        return this[common_1.rawDataSymbol].unmod.user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return await this._client.users.getUserById(this[common_1.rawDataSymbol].unmod.user_id);
    }
};
exports.EventSubChannelUnmodModerationEvent = EventSubChannelUnmodModerationEvent;
exports.EventSubChannelUnmodModerationEvent = EventSubChannelUnmodModerationEvent = tslib_1.__decorate([
    (0, common_1.rtfm)('eventsub-base', 'EventSubChannelUnmodModerationEvent', 'broadcasterId')
], EventSubChannelUnmodModerationEvent);
