// @twurple/eventsub-base/lib/events/moderation/EventSubChannelUnraidModerationEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventSubChannelUnraidModerationEvent = void 0;
const tslib_1 = require("tslib");
const common_1 = require("@twurple/common");
const EventSubChannelBaseModerationEvent_1 = require("./EventSubChannelBaseModerationEvent");
/**
 * An EventSub event representing a moderator canceling the raid on a channel.
 */
let EventSubChannelUnraidModerationEvent = class EventSubChannelUnraidModerationEvent extends EventSubChannelBaseModerationEvent_1.EventSubChannelBaseModerationEvent {
    constructor() {
        super(...arguments);
        this.moderationAction = 'unraid';
    }
    /**
     * The ID of the user no longer being raided.
     */
    get userId() {
        return this[common_1.rawDataSymbol].unraid.user_id;
    }
    /**
     * The name of the user no longer being raided.
     */
    get userName() {
        return this[common_1.rawDataSymbol].unraid.user_login;
    }
    /**
     * The display name of the user no longer being raided.
     */
    get userDisplayName() {
        return this[common_1.rawDataSymbol].unraid.user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return await this._client.users.getUserById(this[common_1.rawDataSymbol].unraid.user_id);
    }
};
exports.EventSubChannelUnraidModerationEvent = EventSubChannelUnraidModerationEvent;
exports.EventSubChannelUnraidModerationEvent = EventSubChannelUnraidModerationEvent = tslib_1.__decorate([
    (0, common_1.rtfm)('eventsub-base', 'EventSubChannelUnraidModerationEvent', 'broadcasterId')
], EventSubChannelUnraidModerationEvent);
