// @twurple/eventsub-base/lib/events/moderation/EventSubChannelBanModerationEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventSubChannelBanModerationEvent = void 0;
const tslib_1 = require("tslib");
const common_1 = require("@twurple/common");
const EventSubChannelBaseModerationEvent_1 = require("./EventSubChannelBaseModerationEvent");
/**
 * An EventSub event representing a moderator banning a user on a channel.
 */
let EventSubChannelBanModerationEvent = class EventSubChannelBanModerationEvent extends EventSubChannelBaseModerationEvent_1.EventSubChannelBaseModerationEvent {
    constructor() {
        super(...arguments);
        this.moderationAction = 'ban';
    }
    /**
     * The ID of the user being banned.
     */
    get userId() {
        return this[common_1.rawDataSymbol].ban.user_id;
    }
    /**
     * The name of the user being banned.
     */
    get userName() {
        return this[common_1.rawDataSymbol].ban.user_login;
    }
    /**
     * The display name of the user being banned.
     */
    get userDisplayName() {
        return this[common_1.rawDataSymbol].ban.user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return await this._client.users.getUserById(this[common_1.rawDataSymbol].ban.user_id);
    }
    /**
     * The reason given for the ban.
     */
    get reason() {
        return this[common_1.rawDataSymbol].ban.reason;
    }
};
exports.EventSubChannelBanModerationEvent = EventSubChannelBanModerationEvent;
exports.EventSubChannelBanModerationEvent = EventSubChannelBanModerationEvent = tslib_1.__decorate([
    (0, common_1.rtfm)('eventsub-base', 'EventSubChannelBanModerationEvent', 'broadcasterId')
], EventSubChannelBanModerationEvent);
