// @twurple/eventsub-base/lib/events/moderation/EventSubChannelVipModerationEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventSubChannelVipModerationEvent = void 0;
const tslib_1 = require("tslib");
const common_1 = require("@twurple/common");
const EventSubChannelBaseModerationEvent_1 = require("./EventSubChannelBaseModerationEvent");
/**
 * An EventSub event representing a user having gained VIP status on a channel.
 */
let EventSubChannelVipModerationEvent = class EventSubChannelVipModerationEvent extends EventSubChannelBaseModerationEvent_1.EventSubChannelBaseModerationEvent {
    constructor() {
        super(...arguments);
        this.moderationAction = 'vip';
    }
    /**
     * The ID of the user gaining VIP status.
     */
    get userId() {
        return this[common_1.rawDataSymbol].vip.user_id;
    }
    /**
     * The name of the user gaining VIP status.
     */
    get userName() {
        return this[common_1.rawDataSymbol].vip.user_login;
    }
    /**
     * The display name of the user gaining VIP status.
     */
    get userDisplayName() {
        return this[common_1.rawDataSymbol].vip.user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return await this._client.users.getUserById(this[common_1.rawDataSymbol].vip.user_id);
    }
};
exports.EventSubChannelVipModerationEvent = EventSubChannelVipModerationEvent;
exports.EventSubChannelVipModerationEvent = EventSubChannelVipModerationEvent = tslib_1.__decorate([
    (0, common_1.rtfm)('eventsub-base', 'EventSubChannelVipModerationEvent', 'broadcasterId')
], EventSubChannelVipModerationEvent);
