// @twurple/eventsub-base/es/events/moderation/EventSubChannelSharedChatUnbanModerationEvent.mjs
import { __decorate } from "tslib";
import { rawDataSymbol, rtfm } from '@twurple/common';
import { EventSubChannelBaseModerationEvent } from "./EventSubChannelBaseModerationEvent.mjs";
/**
 * An EventSub event representing a moderator unbanning a user in another channel during a shared chat session.
 */
let EventSubChannelSharedChatUnbanModerationEvent = class EventSubChannelSharedChatUnbanModerationEvent extends EventSubChannelBaseModerationEvent {
    constructor() {
        super(...arguments);
        this.moderationAction = 'shared_chat_unban';
    }
    /**
     * The ID of the user being unbanned.
     */
    get userId() {
        return this[rawDataSymbol].shared_chat_unban.user_id;
    }
    /**
     * The name of the user being unbanned.
     */
    get userName() {
        return this[rawDataSymbol].shared_chat_unban.user_login;
    }
    /**
     * The display name of the user being unbanned.
     */
    get userDisplayName() {
        return this[rawDataSymbol].shared_chat_unban.user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return await this._client.users.getUserById(this[rawDataSymbol].shared_chat_unban.user_id);
    }
};
EventSubChannelSharedChatUnbanModerationEvent = __decorate([
    rtfm('eventsub-base', 'EventSubChannelSharedChatUnbanModerationEvent', 'broadcasterId')
], EventSubChannelSharedChatUnbanModerationEvent);
export { EventSubChannelSharedChatUnbanModerationEvent };
