// @twurple/eventsub-base/es/events/moderation/EventSubChannelVipModerationEvent.mjs
import { __decorate } from "tslib";
import { rawDataSymbol, rtfm } from '@twurple/common';
import { EventSubChannelBaseModerationEvent } from "./EventSubChannelBaseModerationEvent.mjs";
/**
 * An EventSub event representing a user having gained VIP status on a channel.
 */
let EventSubChannelVipModerationEvent = class EventSubChannelVipModerationEvent extends EventSubChannelBaseModerationEvent {
    constructor() {
        super(...arguments);
        this.moderationAction = 'vip';
    }
    /**
     * The ID of the user gaining VIP status.
     */
    get userId() {
        return this[rawDataSymbol].vip.user_id;
    }
    /**
     * The name of the user gaining VIP status.
     */
    get userName() {
        return this[rawDataSymbol].vip.user_login;
    }
    /**
     * The display name of the user gaining VIP status.
     */
    get userDisplayName() {
        return this[rawDataSymbol].vip.user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return await this._client.users.getUserById(this[rawDataSymbol].vip.user_id);
    }
};
EventSubChannelVipModerationEvent = __decorate([
    rtfm('eventsub-base', 'EventSubChannelVipModerationEvent', 'broadcasterId')
], EventSubChannelVipModerationEvent);
export { EventSubChannelVipModerationEvent };
