// @twurple/eventsub-base/es/events/moderation/EventSubChannelDeleteModerationEvent.mjs
import { __decorate } from "tslib";
import { rawDataSymbol, rtfm } from '@twurple/common';
import { EventSubChannelBaseModerationEvent } from "./EventSubChannelBaseModerationEvent.mjs";
/**
 * An EventSub event representing a moderator deleting a message on a channel.
 */
let EventSubChannelDeleteModerationEvent = class EventSubChannelDeleteModerationEvent extends EventSubChannelBaseModerationEvent {
    constructor() {
        super(...arguments);
        this.moderationAction = 'delete';
    }
    /**
     * The ID of the user whose message is being deleted.
     */
    get userId() {
        return this[rawDataSymbol].delete.user_id;
    }
    /**
     * The name of the user whose message is being deleted.
     */
    get userName() {
        return this[rawDataSymbol].delete.user_login;
    }
    /**
     * The display name of the user whose message is being deleted.
     */
    get userDisplayName() {
        return this[rawDataSymbol].delete.user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return await this._client.users.getUserById(this[rawDataSymbol].delete.user_id);
    }
    /**
     * The ID of the message being deleted.
     */
    get messageId() {
        return this[rawDataSymbol].delete.message_id;
    }
    /**
     * The message text of the message being deleted.
     */
    get messageText() {
        return this[rawDataSymbol].delete.message_body;
    }
};
EventSubChannelDeleteModerationEvent = __decorate([
    rtfm('eventsub-base', 'EventSubChannelDeleteModerationEvent', 'broadcasterId')
], EventSubChannelDeleteModerationEvent);
export { EventSubChannelDeleteModerationEvent };
