// @twurple/eventsub-base/es/events/moderation/EventSubChannelUnbanRequestModerationEvent.mjs
import { __decorate } from "tslib";
import { rawDataSymbol, rtfm } from '@twurple/common';
import { EventSubChannelBaseModerationEvent } from "./EventSubChannelBaseModerationEvent.mjs";
/**
 * An EventSub event representing a moderator resolving an unban request on a channel.
 */
let EventSubChannelUnbanRequestModerationEvent = class EventSubChannelUnbanRequestModerationEvent extends EventSubChannelBaseModerationEvent {
    /** @internal */
    constructor(data, action, client) {
        super(data, client);
        this.moderationAction = action;
    }
    /**
     * Whether the unban request was approved or denied.
     */
    get isApproved() {
        return this[rawDataSymbol].unban_request.is_approved;
    }
    /**
     * The ID of the banned user.
     */
    get userId() {
        return this[rawDataSymbol].unban_request.user_id;
    }
    /**
     * The name of the banned user.
     */
    get userName() {
        return this[rawDataSymbol].unban_request.user_login;
    }
    /**
     * The display name of the banned user.
     */
    get userDisplayName() {
        return this[rawDataSymbol].unban_request.user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return await this._client.users.getUserById(this[rawDataSymbol].unban_request.user_id);
    }
    /**
     * The message included by the moderator explaining their approval or denial.
     */
    get moderatorMessage() {
        return this[rawDataSymbol].unban_request.moderator_message;
    }
};
EventSubChannelUnbanRequestModerationEvent = __decorate([
    rtfm('eventsub-base', 'EventSubChannelUnbanRequestModerationEvent', 'broadcasterId')
], EventSubChannelUnbanRequestModerationEvent);
export { EventSubChannelUnbanRequestModerationEvent };
