// @twurple/eventsub-base/es/events/moderation/EventSubChannelRaidModerationEvent.mjs
import { __decorate } from "tslib";
import { rawDataSymbol, rtfm } from '@twurple/common';
import { EventSubChannelBaseModerationEvent } from "./EventSubChannelBaseModerationEvent.mjs";
/**
 * An EventSub event representing a moderator starting a raid on a channel.
 */
let EventSubChannelRaidModerationEvent = class EventSubChannelRaidModerationEvent extends EventSubChannelBaseModerationEvent {
    constructor() {
        super(...arguments);
        this.moderationAction = 'raid';
    }
    /**
     * The ID of the user being raided.
     */
    get userId() {
        return this[rawDataSymbol].raid.user_id;
    }
    /**
     * The name of the user being raided.
     */
    get userName() {
        return this[rawDataSymbol].raid.user_login;
    }
    /**
     * The display name of the user being raided.
     */
    get userDisplayName() {
        return this[rawDataSymbol].raid.user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return await this._client.users.getUserById(this[rawDataSymbol].raid.user_id);
    }
    /**
     * The number of viewers who came with the raid.
     */
    get viewerCount() {
        return this[rawDataSymbol].raid.viewer_count;
    }
};
EventSubChannelRaidModerationEvent = __decorate([
    rtfm('eventsub-base', 'EventSubChannelRaidModerationEvent', 'broadcasterId')
], EventSubChannelRaidModerationEvent);
export { EventSubChannelRaidModerationEvent };
