// @twurple/eventsub-base/es/events/moderation/EventSubChannelUnvipModerationEvent.mjs
import { __decorate } from "tslib";
import { rawDataSymbol, rtfm } from '@twurple/common';
import { EventSubChannelBaseModerationEvent } from "./EventSubChannelBaseModerationEvent.mjs";
/**
 * An EventSub event representing a user having lost VIP status on a channel.
 */
let EventSubChannelUnvipModerationEvent = class EventSubChannelUnvipModerationEvent extends EventSubChannelBaseModerationEvent {
    constructor() {
        super(...arguments);
        this.moderationAction = 'unvip';
    }
    /**
     * The ID of the user losing VIP status.
     */
    get userId() {
        return this[rawDataSymbol].unvip.user_id;
    }
    /**
     * The name of the user losing VIP status.
     */
    get userName() {
        return this[rawDataSymbol].unvip.user_login;
    }
    /**
     * The display name of the user losing VIP status.
     */
    get userDisplayName() {
        return this[rawDataSymbol].unvip.user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return await this._client.users.getUserById(this[rawDataSymbol].unvip.user_id);
    }
};
EventSubChannelUnvipModerationEvent = __decorate([
    rtfm('eventsub-base', 'EventSubChannelUnvipModerationEvent', 'broadcasterId')
], EventSubChannelUnvipModerationEvent);
export { EventSubChannelUnvipModerationEvent };
