// @twurple/eventsub-base/es/events/moderation/EventSubChannelUnbanModerationEvent.mjs
import { __decorate } from "tslib";
import { rawDataSymbol, rtfm } from '@twurple/common';
import { EventSubChannelBaseModerationEvent } from "./EventSubChannelBaseModerationEvent.mjs";
/**
 * An EventSub event representing a moderator unbanning a user on a channel.
 */
let EventSubChannelUnbanModerationEvent = class EventSubChannelUnbanModerationEvent extends EventSubChannelBaseModerationEvent {
    constructor() {
        super(...arguments);
        this.moderationAction = 'unban';
    }
    /**
     * The ID of the user being unbanned.
     */
    get userId() {
        return this[rawDataSymbol].unban.user_id;
    }
    /**
     * The name of the user being unbanned.
     */
    get userName() {
        return this[rawDataSymbol].unban.user_login;
    }
    /**
     * The display name of the user being unbanned.
     */
    get userDisplayName() {
        return this[rawDataSymbol].unban.user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return await this._client.users.getUserById(this[rawDataSymbol].unban.user_id);
    }
};
EventSubChannelUnbanModerationEvent = __decorate([
    rtfm('eventsub-base', 'EventSubChannelUnbanModerationEvent', 'broadcasterId')
], EventSubChannelUnbanModerationEvent);
export { EventSubChannelUnbanModerationEvent };
