// @twurple/eventsub-base/lib/events/EventSubAutoModMessageHoldEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventSubAutoModMessageHoldEvent = void 0;
const tslib_1 = require("tslib");
const shared_utils_1 = require("@d-fischer/shared-utils");
const common_1 = require("@twurple/common");
/**
 * An EventSub event representing chat message being held by AutoMod in a channel.
 */
let EventSubAutoModMessageHoldEvent = class EventSubAutoModMessageHoldEvent extends common_1.DataObject {
    /** @internal */
    constructor(data, client) {
        super(data);
        this._client = client;
    }
    /**
     * The ID of the broadcaster in whose channel the message has been held by AutoMod.
     */
    get broadcasterId() {
        return this[common_1.rawDataSymbol].broadcaster_user_id;
    }
    /**
     * The name of the broadcaster in whose channel the message has been held by AutoMod.
     */
    get broadcasterName() {
        return this[common_1.rawDataSymbol].broadcaster_user_login;
    }
    /**
     * The display name of the broadcaster in whose channel the message has been held by AutoMod.
     */
    get broadcasterDisplayName() {
        return this[common_1.rawDataSymbol].broadcaster_user_name;
    }
    /**
     * Gets more information about the broadcaster.
     */
    async getBroadcaster() {
        return (0, common_1.checkRelationAssertion)(await this._client.users.getUserById(this[common_1.rawDataSymbol].broadcaster_user_id));
    }
    /**
     * The ID of the user whose message has been held by AutoMod.
     */
    get userId() {
        return this[common_1.rawDataSymbol].user_id;
    }
    /**
     * The name of the user whose message has been held by AutoMod.
     */
    get userName() {
        return this[common_1.rawDataSymbol].user_login;
    }
    /**
     * The display name of the user whose message has been held by AutoMod.
     */
    get userDisplayName() {
        return this[common_1.rawDataSymbol].user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return (0, common_1.checkRelationAssertion)(await this._client.users.getUserById(this[common_1.rawDataSymbol].user_id));
    }
    /**
     * The ID of the chat message held by AutoMod.
     */
    get messageId() {
        return this[common_1.rawDataSymbol].message_id;
    }
    /**
     * The plain text of the message.
     */
    get messageText() {
        return this[common_1.rawDataSymbol].message.text;
    }
    /**
     * The pre-parsed message parts.
     */
    get messageParts() {
        return this[common_1.rawDataSymbol].message.fragments;
    }
    /**
     * The category of the message.
     */
    get category() {
        return this[common_1.rawDataSymbol].category;
    }
    /**
     * The level of severity.
     */
    get level() {
        return this[common_1.rawDataSymbol].level;
    }
    /**
     * The date of when AutoMod held the message.
     */
    get holdDate() {
        return new Date(this[common_1.rawDataSymbol].held_at);
    }
};
exports.EventSubAutoModMessageHoldEvent = EventSubAutoModMessageHoldEvent;
tslib_1.__decorate([
    (0, shared_utils_1.Enumerable)(false)
], EventSubAutoModMessageHoldEvent.prototype, "_client", void 0);
exports.EventSubAutoModMessageHoldEvent = EventSubAutoModMessageHoldEvent = tslib_1.__decorate([
    (0, common_1.rtfm)('eventsub-base', 'EventSubAutoModMessageHoldEvent', 'messageId')
], EventSubAutoModMessageHoldEvent);
