// @twurple/eventsub-base/es/events/EventSubAutoModMessageHoldEvent.mjs
import { __decorate } from "tslib";
import { Enumerable } from '@d-fischer/shared-utils';
import { checkRelationAssertion, DataObject, rawDataSymbol, rtfm } from '@twurple/common';
/**
 * An EventSub event representing chat message being held by AutoMod in a channel.
 */
let EventSubAutoModMessageHoldEvent = class EventSubAutoModMessageHoldEvent extends DataObject {
    /** @internal */
    constructor(data, client) {
        super(data);
        this._client = client;
    }
    /**
     * The ID of the broadcaster in whose channel the message has been held by AutoMod.
     */
    get broadcasterId() {
        return this[rawDataSymbol].broadcaster_user_id;
    }
    /**
     * The name of the broadcaster in whose channel the message has been held by AutoMod.
     */
    get broadcasterName() {
        return this[rawDataSymbol].broadcaster_user_login;
    }
    /**
     * The display name of the broadcaster in whose channel the message has been held by AutoMod.
     */
    get broadcasterDisplayName() {
        return this[rawDataSymbol].broadcaster_user_name;
    }
    /**
     * Gets more information about the broadcaster.
     */
    async getBroadcaster() {
        return checkRelationAssertion(await this._client.users.getUserById(this[rawDataSymbol].broadcaster_user_id));
    }
    /**
     * The ID of the user whose message has been held by AutoMod.
     */
    get userId() {
        return this[rawDataSymbol].user_id;
    }
    /**
     * The name of the user whose message has been held by AutoMod.
     */
    get userName() {
        return this[rawDataSymbol].user_login;
    }
    /**
     * The display name of the user whose message has been held by AutoMod.
     */
    get userDisplayName() {
        return this[rawDataSymbol].user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return checkRelationAssertion(await this._client.users.getUserById(this[rawDataSymbol].user_id));
    }
    /**
     * The ID of the chat message held by AutoMod.
     */
    get messageId() {
        return this[rawDataSymbol].message_id;
    }
    /**
     * The plain text of the message.
     */
    get messageText() {
        return this[rawDataSymbol].message.text;
    }
    /**
     * The pre-parsed message parts.
     */
    get messageParts() {
        return this[rawDataSymbol].message.fragments;
    }
    /**
     * The category of the message.
     */
    get category() {
        return this[rawDataSymbol].category;
    }
    /**
     * The level of severity.
     */
    get level() {
        return this[rawDataSymbol].level;
    }
    /**
     * The date of when AutoMod held the message.
     */
    get holdDate() {
        return new Date(this[rawDataSymbol].held_at);
    }
};
__decorate([
    Enumerable(false)
], EventSubAutoModMessageHoldEvent.prototype, "_client", void 0);
EventSubAutoModMessageHoldEvent = __decorate([
    rtfm('eventsub-base', 'EventSubAutoModMessageHoldEvent', 'messageId')
], EventSubAutoModMessageHoldEvent);
export { EventSubAutoModMessageHoldEvent };
