// @twurple/eventsub-base/es/events/EventSubAutoModMessageUpdateEvent.mjs
import { __decorate } from "tslib";
import { Enumerable } from '@d-fischer/shared-utils';
import { checkRelationAssertion, DataObject, rawDataSymbol, rtfm } from '@twurple/common';
/**
 * An EventSub event representing a held chat message by AutoMod being resolved in a channel.
 */
let EventSubAutoModMessageUpdateEvent = class EventSubAutoModMessageUpdateEvent extends DataObject {
    /** @internal */
    constructor(data, client) {
        super(data);
        this._client = client;
    }
    /**
     * The ID of the broadcaster in whose channel the held message was resolved.
     */
    get broadcasterId() {
        return this[rawDataSymbol].broadcaster_user_id;
    }
    /**
     * The name of the broadcaster in whose channel the held message was resolved.
     */
    get broadcasterName() {
        return this[rawDataSymbol].broadcaster_user_login;
    }
    /**
     * The display name of the broadcaster in channel the held message was resolved.
     */
    get broadcasterDisplayName() {
        return this[rawDataSymbol].broadcaster_user_name;
    }
    /**
     * Gets more information about the broadcaster.
     */
    async getBroadcaster() {
        return checkRelationAssertion(await this._client.users.getUserById(this[rawDataSymbol].broadcaster_user_id));
    }
    /**
     * The ID of the moderator who resolved the held message.
     */
    get moderatorId() {
        return this[rawDataSymbol].user_id;
    }
    /**
     * The name of the moderator who resolved the held message.
     */
    get moderatorName() {
        return this[rawDataSymbol].user_login;
    }
    /**
     * The display name of the moderator who resolved the held message.
     */
    get moderatorDisplayName() {
        return this[rawDataSymbol].user_name;
    }
    /**
     * Gets more information about the moderator.
     */
    async getModerator() {
        return checkRelationAssertion(await this._client.users.getUserById(this[rawDataSymbol].moderator_user_id));
    }
    /**
     * The ID of the user whose message was held by AutoMod.
     */
    get userId() {
        return this[rawDataSymbol].user_id;
    }
    /**
     * The name of the user whose message was held by AutoMod.
     */
    get userName() {
        return this[rawDataSymbol].user_login;
    }
    /**
     * The display name of the user whose message was held by AutoMod.
     */
    get userDisplayName() {
        return this[rawDataSymbol].user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return checkRelationAssertion(await this._client.users.getUserById(this[rawDataSymbol].user_id));
    }
    /**
     * The ID of the message held by AutoMod.
     */
    get messageId() {
        return this[rawDataSymbol].message_id;
    }
    /**
     * The plain text of the message.
     */
    get messageText() {
        return this[rawDataSymbol].message.text;
    }
    /**
     * The pre-parsed message parts.
     */
    get messageParts() {
        return this[rawDataSymbol].message.fragments;
    }
    /**
     * The category of the message.
     */
    get category() {
        return this[rawDataSymbol].category;
    }
    /**
     * The level of severity.
     */
    get level() {
        return this[rawDataSymbol].level;
    }
    /**
     * The status of the resolved message.
     */
    get status() {
        return this[rawDataSymbol].status;
    }
    /**
     * The date of when AutoMod held the message.
     */
    get holdDate() {
        return new Date(this[rawDataSymbol].held_at);
    }
};
__decorate([
    Enumerable(false)
], EventSubAutoModMessageUpdateEvent.prototype, "_client", void 0);
EventSubAutoModMessageUpdateEvent = __decorate([
    rtfm('eventsub-base', 'EventSubAutoModMessageUpdateEvent', 'messageId')
], EventSubAutoModMessageUpdateEvent);
export { EventSubAutoModMessageUpdateEvent };
