// @twurple/eventsub-base/es/events/EventSubChannelBitsUseEvent.mjs
import { __decorate } from "tslib";
import { Enumerable, mapNullable } from '@d-fischer/shared-utils';
import { checkRelationAssertion, DataObject, rawDataSymbol, rtfm } from '@twurple/common';
import { EventSubChannelBitsUsePowerUp } from "./common/EventSubChannelBitsUsePowerUp.mjs";
/**
 * An EventSub event representing bits being used in a channel.
 */
let EventSubChannelBitsUseEvent = class EventSubChannelBitsUseEvent extends DataObject {
    /** @internal */
    constructor(data, client) {
        super(data);
        this._client = client;
    }
    /**
     * The ID of the broadcaster in whose channel the bits were used.
     */
    get broadcasterId() {
        return this[rawDataSymbol].broadcaster_user_id;
    }
    /**
     * The name of the broadcaster in whose channel the bits were used.
     */
    get broadcasterName() {
        return this[rawDataSymbol].broadcaster_user_login;
    }
    /**
     * The display name of the broadcaster in whose channel the bits were used.
     */
    get broadcasterDisplayName() {
        return this[rawDataSymbol].broadcaster_user_name;
    }
    /**
     * Gets more information about the broadcaster.
     */
    async getBroadcaster() {
        return checkRelationAssertion(await this._client.users.getUserById(this[rawDataSymbol].broadcaster_user_id));
    }
    /**
     * The ID of the user who has used bits.
     */
    get userId() {
        return this[rawDataSymbol].user_id;
    }
    /**
     * The name of the user who has used bits.
     */
    get userName() {
        return this[rawDataSymbol].user_login;
    }
    /**
     * The display name of the user who has used bits.
     */
    get userDisplayName() {
        return this[rawDataSymbol].user_name;
    }
    /**
     * Gets more information about the user.
     */
    async getUser() {
        return checkRelationAssertion(await this._client.users.getUserById(this[rawDataSymbol].user_id));
    }
    /**
     * The type of the bits usage.
     */
    get type() {
        return this[rawDataSymbol].type;
    }
    /**
     * The number of bits used.
     */
    get bits() {
        return this[rawDataSymbol].bits;
    }
    /**
     * The chat message in plain text, or `null` if it's not applicable.
     */
    get messageText() {
        var _a, _b;
        return (_b = (_a = this[rawDataSymbol].message) === null || _a === void 0 ? void 0 : _a.text) !== null && _b !== void 0 ? _b : null;
    }
    /**
     * Ordered list of chat message fragments, or `null` if it's not applicable.
     */
    get messageParts() {
        var _a, _b;
        return (_b = (_a = this[rawDataSymbol].message) === null || _a === void 0 ? void 0 : _a.fragments) !== null && _b !== void 0 ? _b : null;
    }
    /**
     * The Power-up data, or `null` if it's not applicable.
     */
    get powerUp() {
        return mapNullable(this[rawDataSymbol].power_up, v => new EventSubChannelBitsUsePowerUp(v));
    }
};
__decorate([
    Enumerable(false)
], EventSubChannelBitsUseEvent.prototype, "_client", void 0);
EventSubChannelBitsUseEvent = __decorate([
    rtfm('eventsub-base', 'EventSubChannelBitsUserEvent', 'userId')
], EventSubChannelBitsUseEvent);
export { EventSubChannelBitsUseEvent };
