// @twurple/eventsub-base/lib/events/chatNotifications/EventSubChannelChatSharedChatCommunitySubGiftNotificationEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventSubChannelChatSharedChatCommunitySubGiftNotificationEvent = void 0;
const tslib_1 = require("tslib");
const common_1 = require("@twurple/common");
const EventSubChannelChatBaseNotificationEvent_1 = require("./EventSubChannelChatBaseNotificationEvent");
/**
 * An EventSub event representing a community sub gift notification in another channel's chat during a shared chat
 * session.
 */
let EventSubChannelChatSharedChatCommunitySubGiftNotificationEvent = class EventSubChannelChatSharedChatCommunitySubGiftNotificationEvent extends EventSubChannelChatBaseNotificationEvent_1.EventSubChannelChatBaseNotificationEvent {
    constructor() {
        super(...arguments);
        this.type = 'shared_chat_community_sub_gift';
    }
    /**
     * The ID of the community sub gift.
     */
    get id() {
        return this[common_1.rawDataSymbol].shared_chat_community_sub_gift.id;
    }
    /**
     * The tier of the subscriptions.
     */
    get tier() {
        return this[common_1.rawDataSymbol].shared_chat_community_sub_gift.sub_tier;
    }
    /**
     * The amount of gifts that are part of this community sub gift.
     */
    get amount() {
        return this[common_1.rawDataSymbol].shared_chat_community_sub_gift.total;
    }
    /**
     * The amount of gifts that the gifter has sent in total, or `null` the gift is anonymous.
     */
    get cumulativeAmount() {
        return this[common_1.rawDataSymbol].shared_chat_community_sub_gift.cumulative_total;
    }
};
exports.EventSubChannelChatSharedChatCommunitySubGiftNotificationEvent = EventSubChannelChatSharedChatCommunitySubGiftNotificationEvent;
exports.EventSubChannelChatSharedChatCommunitySubGiftNotificationEvent = EventSubChannelChatSharedChatCommunitySubGiftNotificationEvent = tslib_1.__decorate([
    (0, common_1.rtfm)('eventsub-base', 'EventSubChannelChatSharedChatCommunitySubGiftNotificationEvent', 'broadcasterId')
], EventSubChannelChatSharedChatCommunitySubGiftNotificationEvent);
